pkg/services/sso/redhatsso_service.go

```diff
@@ -290,6 +291,27 @@ func (r *redhatssoService) CreateServiceAccountInternal(accessToken string, requ
 }
 
 func (r *redhatssoService) DeleteServiceAccountInternal(accessToken string, clientId string) *errors.ServiceError {
+	// This is code can be removed once the migrated canary & kas-fleetshard are removed. Performance impact would be less as the number of canary & kas-fleetshard are low.
+	// Once the existing canary or kas-fleetshard are deleted. This logic can be removed.
+	if strings.HasPrefix(clientId, "canary") || strings.HasPrefix(clientId, "kas-fleetshard-agent") {
+		first := 0
+		max := 100
+		for {
+			accounts, err := r.client.GetServiceAccounts(accessToken, first, max)
+			if len(accounts) == 0 {
+				return nil
+			}
+			if err != nil {
+				return errors.NewWithCause(errors.ErrorGeneral, err, "failed to collect internal service accounts")
+			}
```

Comment:
I'm arrived a bit late and this has already been merged, but I believe this is a bug.
If en error happens, `len(accounts)` will be 0 and this function will return nil instead of returning the `error`

The error check should be done before

Reply:
you are right. let me fix it